src/Model/Translatable/TranslatableMethods.php

```diff
@@ -103,8 +103,16 @@ protected function doTranslate($locale = null, $fallbackToDefault = true)
             return $translation;
         }
 
-        if ($fallbackToDefault && $defaultTranslation = $this->findTranslationByLocale($this->getDefaultLocale(), false)) {
-            return $defaultTranslation;
+        if ($fallbackToDefault) {
+            if ($fallbackLocale = $this->computeFallbackLocale($locale)) {
+                if ($translation = $this->findTranslationByLocale($fallbackLocale)) {
```

Comment:
I actually tried that but it then complains about the $fallbackLocale not being defined. And it would make the code less readable too in my opinion.